src/cc/tests/integtest_main.cc

```diff
@@ -14,13 +15,31 @@
 #include "tests/environments/MetaserverEnvironment.h"
 #include "tests/integtest.h"
 
-// TODO(fsareshwala): catch SIGINT and remove kill all environments and remove
-// all files
-
 using namespace KFS::Test;
 using namespace std;
 
 const string QFSTestUtils::kTestHome = "/tmp/qfs";
+static const int kNumChunkServers = 3;
+
+void
+HandleSignal(int signum)
+{
+    cout << "Caught signal: " << strsignal(signum) << endl;
+
+    switch (signum) {
+        case SIGSEGV:
+        case SIGINT:
+            cout << "Terminating environments..." << endl;
+            QFSTest::sMetaserver->TearDown();
+            for (unsigned int i = 0; i < kNumChunkServers; i++) {
+                QFSTest::sChunkservers[i]->TearDown();
```

Comment:
This would be cleaner if we had a static function in QFSTest which calls TearDown() both on sMetaserver and each element in sChunkservers? That function would replace all the lines here.